internal/gh/project_next.go

```diff
@@ -0,0 +1,103 @@
+package gh
+
+import (
+	"context"
+
+	"github.com/shurcooL/githubv4"
+)
+
+// GraphQLFields represent a list of GitHub PNFs (Project Next Field).
+type GraphQLFields struct {
+	TotalCount githubv4.Int
+	Nodes      []GraphQLField
+}
+
+// GraphQLField represents a single GitHub PNF (Project Next Field).
+type GraphQLField struct {
+	ID       githubv4.ID
+	Name     githubv4.String
+	DataType githubv4.String
+	Settings githubv4.String
+}
+
+// GraphQLFieldValues represents list of GitHub PNFVs (Project Next Field Value).
+type GraphQLFieldValues struct {
+	TotalCount githubv4.Int
+	Nodes      []GraphQLFieldValue
+}
+
+// GraphQLFieldValue represents a single GitHub PNFV (Project Next Field Value).
+type GraphQLFieldValue struct {
+	ID           githubv4.ID
+	Value        githubv4.String
+	ProjectField GraphQLField
+
+	// ValueTitle is a special field to display the value in a more human-readable way.
+	ValueTitle string `graphql:"value"`
+}
+
+// GraphQLItems represents a list of GitHub PNIs (Project Next Item).
+type GraphQLItems struct {
+	TotalCount githubv4.Int
+	Nodes      []GraphQLItem
+}
+
+// GraphQLItem represents a single GitHub PNI (Project Next Item).
+type GraphQLItem struct {
+	ID          githubv4.ID
+	FieldValues GraphQLFieldValues `graphql:"fieldValues(first: $fieldsMax)"`
+}
+
+// Querier describes a GitHub GraphQL client that can make a query.
+type Querier interface {
+	Query(ctx context.Context, q interface{}, vars map[string]interface{}) error
+}
+
+// GetPRItems returns the list of PNIs - Project Next Items (cards) associated with the given PR.
+func GetPRItems(client Querier, nodeID string) ([]GraphQLItem, error) {
+	var q struct {
+		Node struct {
+			PullRequest struct {
+				ID                githubv4.String
+				ProjectsNextItems GraphQLItems `graphql:"projectNextItems(first: $itemsMax)"`
+			} `graphql:"... on PullRequest"`
+		} `graphql:"node(id: $nodeID)"`
+	}
+
+	variables := map[string]interface{}{
+		"nodeID":    githubv4.ID(nodeID),
+		"itemsMax":  githubv4.Int(20),
+		"fieldsMax": githubv4.Int(20),
+	}
+
+	err := client.Query(context.Background(), &q, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	if q.Node.PullRequest.ProjectsNextItems.TotalCount == 0 {
+		return []GraphQLItem{}, nil
+	}
+
+	// Set human-readable titles for the values of fields.
+	for _, item := range q.Node.PullRequest.ProjectsNextItems.Nodes {
+		for i, value := range item.FieldValues.Nodes {
+			switch value.ProjectField.DataType {
+			case "ITERATION":
+				item.FieldValues.Nodes[i].ValueTitle, err =
+					GetIterationTitleByID(string(value.Value), string(value.ProjectField.Settings))
+			case "SINGLE_SELECT":
+				item.FieldValues.Nodes[i].ValueTitle, err =
+					GetSingleSelectTitleByID(string(value.Value), string(value.ProjectField.Settings))
```

Comment:
As you can see, there are not any nodes  :)

Reply:
Also, it's Go, so we don't go through all the cases, we exit early enough :)

Reply:
> As you can see, there are not any nodes :)

The point is to care about effectiveness where we can.
If we stop to do it, we get to lightly be ok where is not effective. 
And then when the time comes it would be difficult to find - too many places, and where it was not expected.
The proposal is to keep the mind training. 
Yes, I do remember we do not care about efficiency much for now, but that's an easy change.

Reply:
> > well, I mean developers should be paranoiac about how things can go wrong.
> 
> Sorry, that doesn't sound healthy for either developer or any other human.

Ok, Groves was a successful paranoic. ) 

> Is there any test that I haven't written and that can cover your worries?

no test is needed as we don't have optimizations for now.